lib/trento/activity_logging/activity_catalog.ex

```diff
@@ -0,0 +1,69 @@
+defmodule Trento.ActivityLog.ActivityCatalog do
+  @moduledoc """
+  Activity logging catalog
+  """
+
+  @type logged_activity :: {controller :: module(), activity :: atom()}
+
+  @login_attempt {TrentoWeb.SessionController, :create}
+  @api_key_generation {TrentoWeb.V1.SettingsController, :update_api_key_settings}
```

Comment:
> boundary issue with the `Trento` name space now depending on `TrentoWeb` 
>
I hear you, I see what you mean and I tend to agree.

Consider we already _break_ that dependency direction in projectors, enriching commanded middleware and in an amqp message processor (mainly views and `TrentoWeb.Endpoint` in order to broadcast things via websocket) :see_no_evil: 

Now, I am not saying that since we broke some rule we should ruthlessly keep doing so :smile: 
What I mean is that we accepted the trade-off of having access to `TrentoWeb` components within `Trento` in certain edgy areas (like projectors and what mentioned earlier).

All the references to `TrentoWeb` controllers are kept inside the centralized `Trento.ActivityLog.ActivityCatalog` (meant to be like that) and I would keep accepting this trade-off also for this feature, unless until it matures more.

I will track an item with the concern you're raising.

---

Solution wide, I guess you mean having a top level `./lib/activity_log`  (or similar) next to `./lib/trento` and `./lib/trento_web`. It might make sense indeed, as it would be a cross-cutting concern, as you mentioned.

Gonna track this in the backlog as well.
Thanks!

Reply:
What if we just move the ActivityCatalog to the web app? It may make sense as it refers to controller operations.

Reply:
:thinking: we'd be needing to access `TrentoWeb.{WhateverNamespace}.ActivityCatalog` from within `Trento` and we'd be back in the same situation.

Besides, the activity catalog does belong to activity logging better than web, imho, because it would possibly reference also non-web related activity.